### ISSUE TITLE
NIT: subprocess-run-check / W1510 documentation might be confusing

### ISSUE THREAD

**Issue by nstng** [PyCQA/pylint#7973]
Hi,
https://pylint.pycqa.org/en/latest/user_guide/messages/warning/subprocess-run-check.html states in the example that
```
subprocess.run(["ls"])
```
should be change to
```
subprocess.run(["ls"], check=True)
```
`check=False` is actually the default, i.e., the behavior would be changed. Sure, a developer should verify such changes, but imho this is prone to cause bugs in code where the error handling is done in a wrapper.

Opinions?

Related: https://github.com/PyCQA/pylint/issues/2848

(I was unsure on the correct issue type - let me know if this should be converted.)

**Comment by Pierre-Sassoulas** [comment]
Nice catch the good example should not modify the code behavior.

**Comment by Saumya-ranjan** [comment]
I'd like to take this up. I would really appreciate some clarification on the same

**Comment by Pierre-Sassoulas** [comment]
In the 'good' example check's value should be changed to the default value.

**Comment by turbotimon** [comment]
also, the check is triggered even if the return code is already being checked. e.g.
```python
    p = subprocess.run(
        "echo 'hi'",
        capture_output=True,
        shell=True,
    ) 
    if code := p.returncode:
        # error handling
```

**Comment by turbotimon** [comment]
I think if you they would implement subprocess from scratch, they would choose `check=True` as the default. But since it was not, they can not just change the default because it would break a lot of code.

But i agree, that the 'correct code' example is a bit aggressive since this would be also a "correct example" i suppose:
```python
    p = subprocess.run(
        "echo 'hi'",
        capture_output=True,
        shell=True,
        check=False,
    ) 
    if p.returncode: # not 0
        # error handling
```

**Comment by Pierre-Sassoulas** [comment]
Feel free to review #8097 with the proposed changed and some clarification in the message :)
